src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -164,6 +168,16 @@ class TestSetup:
     __test__ = False
 
 
+_TEST_PATTERN = re.compile(b"def\\s+test_")
+
+
+@rule_helper
+async def _pytest_sources_concurrency(field_set: PythonTestFieldSet) -> int:
+    source_files = await Get(SourceFiles, SourceFilesRequest([field_set.source]))
+    contents = await Get(DigestContents, Digest, source_files.snapshot.digest)
+    return sum([len(_TEST_PATTERN.findall(file.content)) for file in contents])
```

Comment:
This would be good to test. Imo the simplest approach would be to make this last line a plain-and-simple function, then write a normal unit test.